cpp/src/arrow/util/bpacking.cc

```diff
@@ -174,5 +176,221 @@ int unpack32(const uint32_t* in, uint32_t* out, int batch_size, int num_bits) {
 #endif
 }
 
+namespace {
+
+int unpack64_default(const uint8_t* in, uint64_t* out, int batch_size, int num_bits) {
```

Comment:
please comment on how this code was generated?  If there is a script could you add that as well.